db/syncPostsToGrapher.ts

```diff
@@ -23,14 +32,46 @@ const syncPostsToGrapher = async (): Promise<void> => {
     const toDelete = existsInGrapher
         .filter((p) => !doesExistInWordpress[p.id])
         .map((p) => p.id)
+    const replacer = (
+        _match: string,
+        _firstPattern: string,
+        _offset: number,
+        fullString: string,
+        matches: Record<string, string>
+    ): string => {
+        const block = allBlocks[matches["id"].toString()]
+        if (block) return block.post_content
+        else return fullString
+    }
     const toInsert = rows.map((post: any) => {
+        // Now resolve references by using replace with a regex and a function.
+        // There are two special cases - one is that not all refs resolve (🤨)
```

Comment:
the only thing I can think off right now is that a published post is referencing an unpublished / deleted reusable block. Do you think there might be another explanation? Either way, it's nice that this edge case is handled. It probably would be useful surfacing unresolvable reusable blocks as a one-off to authors so they can check for possible content issues.

Reply:
Yes, I think surfacing this to authors is good. They now just stay in the database unmodified so we can do a simple query in datasette and post this for the authors to check those ~5 posts and remove the now broken links to reusable blocks